### DISCUSSION TITLE
LeadingEdgeDevices xsiTEUpper and xsiTELower

### DISCUSSION THREAD

**Discussion by ErikPeetersGKNFokker** [DLR-SL/CPACS#740] [Q&A] [answered]
I will illustrate my question based on the following example from the DC2 CPACS file:
```
<innerBorder>
        <etaLE>0.12</etaLE>
        <xsiTEUpper>0.5</xsiTEUpper>
        <xsiTELower>0.6</xsiTELower>
        <xsiTE>0.2</xsiTE>
        <innerShape>
                <relHeightTE>0.5</relHeightTE>
                <xsiTE>0.7</xsiTE>
        </innerShape>
</innerBorder>
```

This is an innerBorder element of the outerShape of a leadingEdgeDevice as decribed here: https://www.cpacs.de/documentation/CPACS_development/html/e393885e-4e71-dbeb-59f5-1aa281605868.htm

My question is: what do xsiTEUpper and xsiTELower represent? They look similar to the XsiUpperSkin and xsiLowerSkin from the leadingEdgeShape of the trailingEdgeDevice, but they are clearly different. My guess is that they have something to do with the upper and lower skin positions, as with just the xsiTE, there is no way to differentiate between the upper and lower trailing edge position of the leading edge device. This is how I understand the workings of the chordwise (xsi) positions:

![image](https://user-images.githubusercontent.com/76091498/102319702-921f8a80-3f7b-11eb-9abd-2036f08699ee.png)

So how do xsiTEUpper and xsiTELower come in here?

**Comment by MarAlder** [comment, marked as answer]
Hi Erik,
good question, we should improve the documentation here. Your example is not completely valid, though. Let's have a look into the [documentation](https://www.cpacs.de/documentation/CPACS_development/html/e393885e-4e71-dbeb-59f5-1aa281605868.htm):

- First, you have the choice betwee (1) `xsiTE` and (2) the sequence of `xsiTEUppre` and `xsiTELower`. This allows you to choose whether the intersection with the upper and lower profile is simply defined by a straight cut (I think perpendicular to the cord-plane) at `xsiTE`, or by specifying the position on the upper profile by `xsiTEUpper` and on the lower part of the profile by `xsiTELower`. 
- Then you have an additional choice to model the inner shape. Here you might specify a third point by `xsiTE` and `relHeightTE` at this position. 
- The resulting shape of the `innerShape` is usually described using a cubic spline. It is currently implicitly assumed that this spline is tangent to the profile. We should discuss in the community, whether we want to explicitly specify this for CPACS itself or leave it to the geometry processors. 
- In the TiGL 2 implementation `xsiTE` for the innerShape is defined in the control surface coordinate system and orientated from the trailing end to the leading edge. There might be a reason for the orientation, but I'm not really aware of such. @rainman110 @joergbrech do you know the background for this?

So, except for the last point, which I'm not sure about, I would say the parameters behave as follows: 
![temp](https://user-images.githubusercontent.com/43143741/102489124-582eb100-406d-11eb-9901-5ca29e204f25.png)

**Comment by ErikPeetersGKNFokker** [comment]
Hi @MarAlder,

Thank you so much for your thorough reply!

Am I right in saying that the values for xsiTEUpper and xsiTELower in the DC2 example I gave don't make much sense then?

Your explanation of what xsiTEUpper and xsiTELower represent makes sense and I thought it was like that initially, but then when I saw the values attached to them in the DC2 CPACS file I started doubting that.

Cheers,
Erik

**Comment by ErikPeetersGKNFokker** [comment]
Hi @MarAlder,

I have been working on modelling the trailing edge shape of the leading edge devices, but I run into a problem:
None of the CPACS files that I have been able to find have the TE shape of the LED defined properly. 
I have found one where the innerShape is defined, but that is the one I mentioned before in this thread, where the locations of the xsiTEUpper and xsiTELower make no sense.
I have found a couple of other CPACS files that use a leadingEdgeShape in the leading edge device. For some reason, this is a possibility according to the documentation:
![image](https://user-images.githubusercontent.com/76091498/103658046-fd50f100-4f6a-11eb-9237-16d1d9a175c7.png)
But this makes absolutely no sense to me, because why would you define the leading edge shape on the leading edge device?

My question to you is then; do you have a CPACS file where the innerShape of the LED is properly defined?

Kind regards,
Erik Peeters

**Comment by MarAlder** [comment]
Hi @ErikPeetersGKNFokker ,

the [leadingEdgeShape](https://www.cpacs.de/documentation/CPACS_3_2_0_Docs/html/1a74628c-9acc-3f30-eec6-4d3e47da995b.htm) makes (in this context) no sense for me as well. I'll try to find out where it comes from, but I guess it has never been used.

If you contact me at cpacs@dlr.de we can exchange some files.
